modules/checkbox/react/spec/Checkbox.spec.tsx

```diff
@@ -1,102 +1,134 @@
+        const {getByLabelText} = render(
+          <Checkbox defaultChecked={true} onChange={cb} label={label} />
+        );
+        expect(getByLabelText(label)).toHaveProperty('checked', true);
+      });
+    });
 
-  test('should call a callback function', () => {
-    const component = mount(<Checkbox onChange={cb} />);
-    const input = component.find('input');
-    input.simulate('change');
+    describe('with defaultChecked=false and checked=true', () => {
+      test('should render a checked checkbox input', () => {
+        const {getByLabelText} = render(
+          <Checkbox defaultChecked={false} checked={true} onChange={cb} label={label} />
+        );
+        expect(getByLabelText(label)).toHaveProperty('checked', true);
+      });
+    });
 
-    expect(cb.mock.calls.length).toBe(1);
-    component.unmount();
-  });
+    describe('with disabled attribute', () => {
+      test('should render a disabled checkbox input', () => {
+        const {getByLabelText} = render(<Checkbox disabled={true} onChange={cb} label={label} />);
+        expect(getByLabelText(label)).toHaveProperty('disabled', true);
+      });
+    });
 
-  test('Checkbox should spread extra props', () => {
-    const component = mount(<Checkbox data-propspread="test" onChange={cb} />);
-    const input = component
-      .find('input') // TODO: Standardize on prop spread location (see #150)
-      .getDOMNode();
-    expect(input.getAttribute('data-propspread')).toBe('test');
-    component.unmount();
-  });
-});
+    test('should pass axe DOM accessibility guidelines', async () => {
+      const html = ReactDOMServer.renderToString(
+        <Checkbox id={'123'} label={'Label'} onChange={jest.fn()} />
+      );
+      expect(await axe(html)).toHaveNoViolations();
+    });
 
-describe('Checkbox Accessibility', () => {
-  test('Checkbox should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <Checkbox id={'123'} label={'Label'} onChange={jest.fn()} />
-    );
-    expect(await axe(html)).toHaveNoViolations();
-  });
+    describe('without a defined id', () => {
+      test('should pass axe DOM accessibility guidelines', async () => {
+        const html = ReactDOMServer.renderToString(
+          <Checkbox label={'Label'} onChange={jest.fn()} />
+        );
+        expect(await axe(html)).toHaveNoViolations();
+      });
+    });
+
+    describe('wrapped in a FormField', () => {
+      test('should pass axe DOM accessibility guidelines', async () => {
+        const html = ReactDOMServer.renderToString(
+          <FormField label="My Field" inputId="my-checkbox-field">
+            <Checkbox disabled={false} checked={true} id="my-checkbox-field" onChange={jest.fn()} />
+          </FormField>
+        );
+        expect(await axe(html)).toHaveNoViolations();
+      });
+    });
+
+    describe('without an id', () => {
+      test('should create a unique id for each instance', async () => {
```

Comment:
Check for the same unique ID if rerendered (https://github.com/Workday/canvas-kit/pull/386/files#diff-c9977ec716fc27e28a4ea0ff50add633R66)